### ISSUE TITLE
scripts: allow list input to run.command

### ISSUE THREAD

**Issue by maxpietsch** [MRtrix3/mrtrix3#928]
I'd suggest allowing to support passing a list of arguments to run.command (on top of the current approach of passing a whitespace separated string that is then split to not break current scripts.).  I have written an example (d4aab3d33561652c4e444d2335dad306a008f87c) of what I was thinking about: It bypasses the `shlex` splitting if the input is a list and bypasses the pipe splitting if the input is a nested list.

This would help for more complicated pipelines such as in `population_template` to allow lists as input to `run.command` where elements of that list can be changed in place - for instance depending on the result of the command. Another benefit of lists is that we don't need to rely on shlex to do the right thing (for instance with funny character encodings).

With those changes, all the commands below would be valid and equivalent:
```
output1 = run.command ('cat /tmp/text | cat')
output2 = run.command (['cat', '/tmp/text', '|', 'cat'])
output3 = run.command ([['cat', '/tmp/text'],['cat']])
```

**Comment by Lestropie** [comment]
Closed by #785.

**Comment by maxpietsch** [comment]
Not too attached to these changes but d4aab3d is not merged into the tagged branch. @Lestropie 
 Just to double check, did you close this on purpose?

**Comment by Lestropie** [comment]
Sorry, thought it was merged in...

**Comment by Lestropie** [comment]
Thinking about this since I have a bunch of Python library changes ongoing that are somewhat related.

@maxpietsch: Is there a strong benefit in supporting specifically usage number 3? If that's supported, then `run.command()` would need to assume that any nested list consists of multiple commands to be executed. I had an idea that would unfortunately be precluded by this: During `population_template`, there are `run.command()` calls that include inputs from all subjects, which end up being very long strings. If a usage such as the following were supported:
```
# Current
# run.command('mrmath ' + ' '.join(allindir('inputs_transformed')) + ' mean initial_template.mif')
# Potential change
run.command([ 'mrmath', allindir('inputs_transformed'), 'mean', 'initial_template.mif' ])
```
(with the output of the `allindir()` function being a list of strings), the terminal output provided by `run.command()` could potentially be changed as thus:
```
# Current
Command:  mrmath inputs_transformed/100307.mif inputs_transformed/100408.mif inputs_transformed/101107.mif inputs_transformed/101309.mif inputs_transformed/101915.mif inputs_transformed/103111.mif inputs_transformed/103414.mif inputs_transformed/103818.mif inputs_transformed/105014.mif inputs_transformed/105115.mif inputs_transformed/106016.mif inputs_transformed/108828.mif inputs_transformed/110411.mif inputs_transformed/111312.mif inputs_transformed/111716.mif inputs_transformed/113619.mif inputs_transformed/113922.mif inputs_transformed/114419.mif inputs_transformed/115320.mif inputs_transformed/116524.mif inputs_transformed/117122.mif inputs_transformed/118528.mif inputs_transformed/118730.mif inputs_transformed/118932.mif inputs_transformed/120111.mif inputs_transformed/122317.mif inputs_transformed/122620.mif inputs_transformed/123117.mif inputs_transformed/123925.mif inputs_transformed/124422.mif inputs_transformed/125525.mif inputs_transformed/126325.mif inputs_transformed/127630.mif inputs_transformed/127933.mif inputs_transformed/128127.mif inputs_transformed/128632.mif inputs_transformed/129028.mif inputs_transformed/130013.mif inputs_transformed/130316.mif inputs_transformed/131217.mif inputs_transformed/131722.mif inputs_transformed/133019.mif inputs_transformed/133928.mif inputs_transformed/135225.mif inputs_transformed/135932.mif inputs_transformed/136833.mif inputs_transformed/138534.mif inputs_transformed/139637.mif inputs_transformed/140925.mif inputs_transformed/144832.mif inputs_transformed/146432.mif inputs_transformed/147737.mif inputs_transformed/148335.mif inputs_transformed/148840.mif inputs_transformed/149337.mif inputs_transformed/149539.mif inputs_transformed/149741.mif inputs_transformed/151223.mif inputs_transformed/151526.mif inputs_transformed/151627.mif inputs_transformed/153025.mif inputs_transformed/154734.mif inputs_transformed/156637.mif inputs_transformed/159340.mif inputs_transformed/160123.mif inputs_transformed/161731.mif inputs_transformed/162733.mif inputs_transformed/163129.mif inputs_transformed/176542.mif inputs_transformed/178950.mif inputs_transformed/188347.mif inputs_transformed/189450.mif inputs_transformed/190031.mif inputs_transformed/192540.mif inputs_transformed/196750.mif inputs_transformed/198451.mif inputs_transformed/199655.mif inputs_transformed/201111.mif inputs_transformed/208226.mif inputs_transformed/211417.mif inputs_transformed/211720.mif inputs_transformed/212318.mif inputs_transformed/214423.mif inputs_transformed/221319.mif inputs_transformed/239944.mif inputs_transformed/245333.mif inputs_transformed/280739.mif inputs_transformed/298051.mif inputs_transformed/366446.mif inputs_transformed/397760.mif inputs_transformed/414229.mif inputs_transformed/499566.mif inputs_transformed/654754.mif inputs_transformed/672756.mif inputs_transformed/751348.mif inputs_transformed/756055.mif inputs_transformed/792564.mif inputs_transformed/856766.mif inputs_transformed/857263.mif inputs_transformed/899885.mif mean initial_template.mif
# Potential change
Command: mrmath [inputs_transformed/*.mif (100 entries)] mean initial_template.mif
```

I also now have `run.command()` supporting `||` and `&&` operators, which would mean that usage 3 would be assuming that the commands represented by nested lists are separated specifically via the piping "`|`" symbol, even though this isn't the only possible usage.

**Comment by Lestropie** [comment]
Closed by #1449.
